### PR TITLE
Maybe make the timer wheel faster

### PR DESCRIPTION
**Pull request by martinthomson**
CHECK THE BENCHMARKS

### CODE REVIEW

**Comment by larseggert** [comment]
@martinthomson could you make clippy happy, so the benches run?

**Comment by larseggert** [comment]
@martinthomson I think we need a microbenchmark or two to see if this makes a difference.

**Comment by martinthomson** [comment]
The microbenchmark I added improves by a third.

File: neqo-crypto/bindings/bindings.toml [hunk 1]
```diff
@@ -265,8 +265,3 @@ enums = [
 [nspr_time]
 types = ["PRTime"]
 functions = ["PR_Now"]
-
-[mozpkix]
-cplusplus = true
-types = ["mozilla::pkix::ErrorCode"]
-enums = ["mozilla::pkix::ErrorCode"]
```
Comment (larseggert) [inline neqo-crypto/bindings/bindings.toml:272]: Seems unrelated?

File: neqo-crypto/src/err.rs [hunk 2]
```diff
@@ -16,7 +16,26 @@ mod codes {
...
+        pub const MOZILLA_PKIX_ERROR_SIGNATURE_ALGORITHM_MISMATCH: Type = -16377;
+        pub const MOZILLA_PKIX_ERROR_OCSP_RESPONSE_FOR_CERT_MISSING: Type = -16376;
+        pub const MOZILLA_PKIX_ERROR_VALIDITY_TOO_LONG: Type = -16375;
+        pub const MOZILLA_PKIX_ERROR_REQUIRED_TLS_FEATURE_MISSING: Type = -16374;
+        pub const MOZILLA_PKIX_ERROR_INVALID_INTEGER_ENCODING: Type = -16373;
+        pub const MOZILLA_PKIX_ERROR_EMPTY_ISSUER_NAME: Type = -16372;
+        pub const MOZILLA_PKIX_ERROR_ADDITIONAL_POLICY_CONSTRAINT_FAILED: Type = -16371;
+        pub const MOZILLA_PKIX_ERROR_SELF_SIGNED_CERT: Type = -16370;
+        pub const MOZILLA_PKIX_ERROR_MITM_DETECTED: Type = -16369;
+    }
```
Comment (larseggert) [inline neqo-crypto/src/err.rs:38]: Seems unrelated? I think you may have based your PR off a different branch than `main`?

**Review by larseggert — CHANGES_REQUESTED** [review]
You probably need to rebase this off `main`.

**Comment by martinthomson** [comment]
```
     Running benches/timer.rs (target/release/deps/timer-acd3fe64661c7be7)
drain a timer quickly   time:   [139.34 ns 148.51 ns 157.44 ns]
                        change: [-71.072% -69.916% -68.575%] (p = 0.00 < 0.05)
                        Performance has improved.
Found 15 outliers among 100 measurements (15.00%)
  12 (12.00%) high mild
  3 (3.00%) high severe
```

That's not a one third improvement, that is this taking one third of the original.  So a more than 3x speed up.

**Review by larseggert — APPROVED** [review]

**Pull request merged**